### ISSUE TITLE
Official CG chatroom (besides w3c IRC channel)

### ISSUE THREAD

**Issue by chicoxyzzy** [WebAssembly/meetings#236]
It will be great to have one official place for general WebAssembly discussions. It feels like IRC is not very popular among developers nowadays (it's pretty popular comparing to different slack workspaces though, some numbers below). Slack requires invites to join so it's not very good option too (though, different Slack workspaces could create [shared channel](https://slackhq.com/introducing-shared-channels-where-you-can-work-with-anyone-in-slack-8c5d2a943f57) for wasm-related discussions). I'm not sure what could be the most optimal application / website for CG discussions, so I propose to discuss this at some of upcoming CG meetings.

cc'ing people who may be interested in this as well @xtuc @dcodeIO @MaxGraey @TheLarkInn @kripken 

Population of WebAssembly channels I'm aware of:
- irc.w3c.org #webassembly channel - 87 members
- Mozilla irc #wasm channel - 70 members
- webpack slack #webassembly channel (about wasm bundling in webpack) - 38 members
- Babel slack #webassembly channel (about [webassemblyjs](https://github.com/xtuc/webassemblyjs) and other tools) - 16 members
- The whole webassemblydevelopers.slack.com (has various channels) - 35 members
- The whole AssemblyScript.slack.com (about [AssemblyScript](https://github.com/AssemblyScript/assemblyscript)) - 20 members
- Russian-speaking https://t.me/webassembly_ru Telegram channel - ~500 members

**Comment by chicoxyzzy** [comment]
Also it may be useful to have channels for CG subgroups

**Comment by chicoxyzzy** [comment]
Emails and GitHub issues are not always so convenient as something chat-like. In fact I think they have different purposes.

**Comment by eholk** [comment]
It may just be because I'm paying more attention to IRC lately, but I feel like there's been an increase in interesting discussion lately on the w3c IRC channel. One big advantage to IRC in my mind is that it's not a proprietary system. That said, most IRC clients could really benefit from improving their UI.

There was [some discussion about this](https://github.com/WebAssembly/design/issues/203) a couple years back. It seems like then the preference was to have substantive design discussions through GitHub issues. Other discussion could happen anywhere, but they should be summarized on the issue tracker if we want to feed them into the WebAssembly design.

It's probably worth looking into setting up logging for the w3c IRC channel.

**Comment by chicoxyzzy** [comment]
Just found WebAssembly group in Spectrum chat

**Comment by binji** [comment]
@chicoxyzzy IIRC you mentioned there's been some progress made by others here, can you summarize? Thanks!

**Comment by chicoxyzzy** [comment]
AFAIK @ttraenkler has created Discord chat recently, also @tschneidereit made some investigations. I think we need to discuss our progress. I think we should discuss ways to collaborate here or plan a call. WDYT?

**Comment by Horcrux7** [comment]
Are there a link to the Discord chat?

**Comment by dcodeIO** [comment]
Fwiw, I do like the ease of use / feature set of Slack when it comes to tech/programming related tasks. Afaik Discord mainly targets gaming related communities (has in-game overlays and whatnot), so not sure if that's a good fit. One alternative I find interesting is https://rocket.chat which aims at being a Slack alternative one can self-host, but I haven't tried it yet. Maybe if someone has a spare server to run it on?

**Comment by devsnek** [comment]
discord also targets open source communities and a good amount use it (yarn, react, elixir, sharex, etc). https://discordapp.com/open-source

**Comment by ttraenkler** [comment]
I am also split minded about Discord. Actually I prefer slack for realtime. Discords excels at voice chat rooms. Also there is spectrum.chat if sou prefer forum like offline chats

**Comment by ttraenkler** [comment]
Here is an invitation to the Discord chat: https://discord.gg/uAtWcX

**Comment by ttraenkler** [comment]
We're currently experimenting (invite only, valid today) how Discord is up to the task and what are the needs of the participants (e.g. separation of concerns, signal to noise, notification settings, rooms, spam, moderation, inclusive atmosphere). 

I would suggest a process of invite only for CG members with real names to keep out trolls and spam so people show their "face" (or at least their real name) when they talk to each other. In my experience with IRC on freenode from the 90s people tend to insult or spam when its all anonymous and also people or rooms might get spammed. 

I would propose we'll gather experience in our experiment over the next weeks and - then the CG can take a better informed decision on what approach to pursue.

**Comment by carlosbaraza** [comment]
Both are pretty good. I am kind of new to using Discord as well, but I found that the email notifications and the web version are pretty good. At the moment I don't have the client installed and I am not missing it, which is pretty impressive.

I'd just say that a product that can win the hearts and communication needs of gamers is probably better than one built with offices in mind. You don't want to piss gamers off, or you'd be out of business soon... o.O

**Comment by sendilkumarn** [comment]
Personally looking for help about WebAssembly and minor nitty-gritty details were a big hassle for me. We will need to make this easy for the newcomers.  Since WebAssembly binds two different communities together, I think most of the people (coming from any background) will definitely have a lot of questions and need initial help to sort things out. 

Discord scales really well and also has a better way to control the noise if any.  Slack is also good but it does not scale when we breach the limits and notifications on Slack is complicated.  

I prefer using Discord and 👍  @ttraenkler for opening the Discord group. Also hearing a lot of good stuff about `spectrum` recently.

**Comment by IsraelOrtuno** [comment]
I really like Discord and it's becoming a very popular alternative to Slack for communities. A few examples:
- Reactflux moved from Slack to Discord (70k+ users) – [Read why](https://reactjs.org/blog/2015/10/19/reactiflux-is-moving-to-discord.html#so-why-discord) 
- Vue.js has its official one called Vue-land (65k+ users): [Visit here](https://vue-land.js.org/)
- Svelte, Vuetify.js, Nuxt.js
- More here: https://discordapp.com/open-source

It's also very interesting for programmers as it has support for markdown, code formatting, loads of integrations...

**Comment by xinbenlv** [comment]
+1 I also like Discord for community. @IsraelOrtuno

**Comment by dtig** [comment]
The WebAssembly discord channel has been up and running for some time now, and has gained adoption, @chicoxyzzy is there more follow to up on here, or can this issue be closed?

**Comment by chicoxyzzy** [comment]
Thank you for the ping @dtig! It can be closed :)

**Comment by ttraenkler** [comment]
The Webassembly Discord was founded in 2018 thanks to the help of people around the Webpack team, the V8 team in Munich, members of Mozilla and NPM just after JSConfEU 2018 in an effort to connect and kick off building an international community and the discussion and organization of conferences. It has been revived in 2019 in an effort to unite the international Webassembly Meetups and while growing slowly initially since the summit and the last CG meeting is kind of  unofficially the "official" Webassembly chatroom and we have crossed 1000 members recently.

As I happen to be the technical "owner" while it has actually been quite a number of people who made it possible to get it off the ground, I feel at this point in time my part in the "mission" to kick off the community by an individual effort is complete. I think it would be the right step if I hand over the ownership of the Discord to the CG, if the CG is interested. If it is welcome, I am open to continue in my role of a technical administrator along with others, but I would like to ensure this becomes part of a neutral body. 

Having IRC experience from the 90s, so far I am pleasantly surprised as how positive the people are on the Discord, so fortunately administration, moderation and code of conduct have not been a noteworthy issue, but I feel no single individual or company should "own" the community - including myself as I feel more and more responsibility in this position and I would like to take action before it becomes more than just a nagging thought in the back of my mind for a long time.

I've been considering proposing this in a CG meeting as discussed with @binji, but I think it does not hurt if we prepare the proposal offline in this issue to evaluate if there is interest in this and what the proposal to vote on should be if any.

**Comment by tlively** [comment]
Update: The CG voted on March 30 to accept stewardship of the WebAssembly Discord ([notes](https://github.com/WebAssembly/meetings/pull/767)).

**Comment by ttraenkler** [comment]
Thanks, I'll coordinate the implementation with @dtig. 👍
